docs/src/libraries/neutron_ic_querier.md

```diff
@@ -0,0 +1,264 @@
+that will shape the overall lifecycle of this library.
+
+This library may take on one of two possible routes with respect to how long it is considered
+active:
+1. single-use, instantiated to perform a particular query and complete
+2. multi-use, instantiated to perform multiple domain queries
+
+For single-use design, one of the main questions is at what point does the query
+actually get registered.
+
+One approach for the single-use approach could be to register the query as soon as
+the library is instantiated. This way, given a complex Valence Program configuration,
+IC Querier library would have the smallest chance of being the bottleneck and
+potentially blocking other libraries from performing their functions (due to query
+result not being fetched yet).
+Alternatively, queries could be registered on-demand. That way, the library
+would be instantiated just like any other library. When a query would be required,
+a message would be executed and the query would get registered.
+
+Both approaches would perform the queries, but the library would remain active.
+For that reason we likely need to introduce some notion of finalization.
+Registered Interchain Queries continue receiving updates according to the `update_period`
+specified during the query registration. This is quite an open design space -
+some potential approaches to query deregistration may involve:
+
+- deregister the query after n query results (e.g. after 1)
+- deregister the query once the result is posted on a block that exceeds a given block/time
+- deregister the query manually with a specific message
+
+## Library in Valence Programs
+
+Neutron IC Querier does not behave as a standard library in that it does not produce
+any fungible outcome. Instead, it produces a foreign type that gets converted
+into a Valence Type.
+
+While that result could be posted directly to the state of this library,
+instead, it is posted to an associated output account meant for storing data.
+Just as some other libraries have a notion of output accounts for transferring some
+funds, Neutron IC Querier has a notion of output account for writing some data.
+
+For example, consider a situation where this library had queried the balance of some
+remote account, parsed the response into a Valence Balance type, and wrote that resulting
+object into its associated output account. That output account may be the input account
+of some other library, which will attempt to perform its function based on the content
+written to its input account. This may involve something along the lines of:
+`if balance > 0, do x; otherwise, do y;`.
+
+It is a little less obvious as to what is the *input account* of Neutron IC Querier.
+For one, the input account could be standard type account that deals with token transfers.
+Prior to registering the interchain query, input account could be expected to receive
+the query deposit (in `untrn`) meant to cover the query registration costs.
```

Comment:
This is not what we do in the ntrn ibc transfer library which also requires a fee. That is the closest analogous library